Apps/W1/HybridGP/app/src/Migration/Support/CheckBooks/GPCheckbookTransactions.table.al

```diff
@@ -216,8 +216,132 @@ table 40101 "GP Checkbook Transactions"
+        LineNum: Integer;
+    begin
+        CreateGeneralJournalBatchIfNeeded(CMTrxType);
+
+        /*  
+            GP CMTrxType we support
+            -- 2 = cash receipt
+            -- 3 = payment
+        */
+        if CMTrxType = 2 then begin
+            DocumentType := DocumentType::" ";
+            JournalTemplateName := 'CASHRCPT';
+            AccountType := AccountType::"G/L Account"
+            /* AccountNo will be GL Account number -- offset account */
+        end else begin
+            DocumentType := DocumentType::Payment;
+            JournalTemplateName := 'PAYMENT';
+            AccountType := AccountType::Vendor;
+            /* AccountNo will be Vendor account number */
+        end;
+
+        GenJournalLineCurrent.SetRange("Journal Template Name", JournalTemplateName);
+        GenJournalLineCurrent.SetRange("Journal Batch Name", 'GPBANK');
+        if GenJournalLineCurrent.FindLast then
+            LineNum := GenJournalLineCurrent."Line No." + 10000
+        else
+            LineNum := 10000;
+
+        GenJournalTemplate.Get(JournalTemplateName);
+
+        GenJournalLine.Init();
+        GenJournalLine.SetHideValidation(true);
+        GenJournalLine.Validate("Posting Date", PostingDate);
+        GenJournalLine.Validate("Document Type", DocumentType);
+        GenJournalLine.Validate("Document No.", DocumentNo);
+        GenJournalLine.Validate("Account Type", AccountType);
+        GenJournalLine.Validate("Account No.", AccountNo);
+        GenJournalLine.Validate(Description, Description);
+        GenJournalLine.Validate(Amount, TrxAmount);
+        GenJournalLine.Validate("Bal. Account Type", GenJournalLine."Bal. Account Type"::"Bank Account");
+        GenJournalLine.Validate("Bal. Account No.", BankAccountNo);
+
+        GenJournalLine.Validate("Source Code", GenJournalTemplate."Source Code");
+        GenJournalLine.Validate("Journal Template Name", JournalTemplateName);
+        GenJournalLine.Validate("Journal Batch Name", 'GPBANK');
+        GenJournalLine.Validate("Line No.", LineNum);
+        GenJournalLine.Validate("Document Date", PostingDate);
+        GenJournalLine.Insert(true);
+    end;
+
+    local procedure CreateGeneralJournalBatchIfNeeded(TrxType: Integer)
+    var
+        GenJournalBatch: Record "Gen. Journal Batch";
+        TemplateName: Code[10];
+    begin
+        GenJournalBatch.SetRange(Name, 'GPBANK');
+
+        if TrxType = 2 then begin
+            GenJournalBatch.SetRange("Journal Template Name", 'CASHRCPT');
+            GenJournalBatch.SetRange("No. Series", 'GJNL-RCPT');
+        end else begin
+            GenJournalBatch.SetRange("Journal Template Name", 'PAYMENT');
+            GenJournalBatch.SetRange("No. Series", 'GJNL-PMT');
+        end;
+
+        if not GenJournalBatch.FindFirst then begin
+            GenJournalBatch.Init();
+            GenJournalBatch.Validate(Name, 'GPBANK');
+
+            if TrxType = 2 then begin
```

Comment:
The Cash Receipt and Payment transaction types could be a variable, and then this could be "if TrxType = CashReceiptTypeId"